packages/contracts-bedrock/test/L2/L1Block.t.sol

```diff
@@ -93,3 +61,57 @@ contract L1BlockTest is CommonTest {
         });
     }
 }
+
+contract L1BlockEcotone_Test is L1BlockTest {
+    /// @dev Tests that setL1BlockValues updates the values appropriately.
+    function testFuzz_setL1BlockValues_succeeds(
+        uint32 basefeeScalar,
+        uint32 blobBasefeeScalar,
+        uint64 sequenceNumber,
+        uint64 timestamp,
+        uint64 number,
+        uint256 basefee,
+        uint256 blobBasefee,
+        bytes32 hash,
+        bytes32 batcherHash
+    )
+        external
+    {
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesEcotone(
+            basefeeScalar, blobBasefeeScalar, sequenceNumber, timestamp, number, basefee, blobBasefee, hash, batcherHash
+        );
+
+        vm.prank(depositor);
+        (bool success,) = address(l1Block).call(functionCallDataPacked);
+        require(success, "Function call failed");
+
+        assertEq(l1Block.basefeeScalar(), basefeeScalar);
+        assertEq(l1Block.blobBasefeeScalar(), blobBasefeeScalar);
+        assertEq(l1Block.sequenceNumber(), sequenceNumber);
+        assertEq(l1Block.timestamp(), timestamp);
+        assertEq(l1Block.number(), number);
+        assertEq(l1Block.basefee(), basefee);
+        assertEq(l1Block.blobBasefee(), blobBasefee);
+        assertEq(l1Block.hash(), hash);
+        assertEq(l1Block.batcherHash(), batcherHash);
+
+		// ensure topmost bits of the sequence number slot are 0
+		bytes32 sequenceSlot = vm.load(address(l1Block), bytes32(uint256(3)));
+		assertEq(uint256(sequenceSlot), uint256(l1Block.sequenceNumber()));
+    }
+
+    /// @dev Tests that `setL1BlockValuesEcotone` fails if address is not depositor 
+    function test_updateValues_succeeds() external {
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesEcotone(
+            type(uint32).max, type(uint32).max, type(uint64).max, type(uint64).max, type(uint64).max, type(uint256).max,
+            type(uint256).max ,bytes32(type(uint256).max), bytes32(type(uint256).max)
+        );
+
+        (bool success,) = address(l1Block).call(functionCallDataPacked);
```

Comment:
We should assert on the returndata too here. Also prefer to use `assertFalse` and `assertTrue` over `require` here

Reply:
Additionally, it would be more consistent with other tests to move the failure case into its own test. ([example](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/test/dispute/DisputeGameFactory.t.sol#L231-L236))

Reply:
fixed